### PR TITLE
fix(snap): update vault to 1.5.4

### PR DESCRIPTION
**Pull request by tonyespy**
This change updates vault to the latest/stable
release (1.5.4), and as such removes the need
to build go 1.12 anymore, so it's been removed
from the snap as well.

Signed-off-by: Tony Espy <espy@canonical.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [**NA**] Tests for the changes have been added (for bug fixes / features)
- [**NA**] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-examples/blob/master/.github/CONTRIBUTING.md.

## What is the current behavior?

This changes has no impact on behavior, it simply updates vault to the latest stable release (1.5.4).

## Issue Number:
**NA**

## What is the new behavior?
NA

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports
Are there any new imports or modules? **No**

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing? No

## Other Information
All of the core & support blackbox tests were run against this change, and tests all pass.

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -492,34 +492,6 @@ parts:
   # from edgex-go which at this point will fail
```
Comment (anonymouse64) [inline snap/snapcraft.yaml:492]: on an unrelated note, this comment can be cleaned up, we don't actually build go anymore so this doesn't apply

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -492,34 +492,6 @@ parts:
...
-      # download the archive, failing on ssl cert problems
-      curl https://dl.google.com/go/$FILE_NAME -O
-      echo "$FILE_HASH $FILE_NAME" > sha256
-      sha256sum -c sha256 | grep OK
-      mkdir -p $SNAPCRAFT_STAGE/go1.12
-      tar -C $SNAPCRAFT_STAGE/go1.12 -xf "$FILE_NAME" --strip-components=1
-    prime:
-      - "-*"
-    # build after kong to prevent staging go binaries into the snap as they
-    # are large - see the TODO on the kong part
```
Comment (anonymouse64) [inline snap/snapcraft.yaml:520]: can you move this comment down to the `after` section on go115 please?

**Review by lenny-intel — APPROVED** [review]
LGTM, based on my limited knowledge... ;-)

**Review by anonymouse64 — COMMENTED** [review]
One nitpick, please keep the comment about part ordering w/ respect to kong and Go

Also, built locally, and installed on my focal desktop and vault starts up fine, so lgtm once the comment has been restored

File: snap/snapcraft.yaml (hunk 1 above)
Comment (tonyespy) [reply snap/snapcraft.yaml]: done

File: snap/snapcraft.yaml (hunk 2 above)
Comment (tonyespy) [reply snap/snapcraft.yaml:520]: and restored

**Review by anonymouse64 — DISMISSED** [review]
LGTM thanks for fixing the comments

**Review by anonymouse64 — APPROVED** [review]

**Pull request merged**